src/app/search/search-filters/sockets.tsx

```diff
@@ -290,6 +290,20 @@ const socketFilters: FilterDefinition[] = [
       (item: DimItem) =>
         item.sockets && compare!(countEnhancedPerks(item.sockets)),
   },
+  {
+    keywords: 'discontinuedperk',
+    description: tl('Filter.DiscontinuedPerk'),
+    destinyVersion: 2,
+    filter: () => (item: DimItem) => {
+      if (!(item.bucket?.sort === 'Weapons' && item.tier.toLowerCase() === 'legendary')) {
+        return false;
+      }
+
+      return getSocketsByCategoryHash(item.sockets, SocketCategoryHashes.WeaponPerks_Reusable).some(
+        (socket) => socket.plugged?.cannotCurrentlyRoll
+      );
```

Comment:
This should not look at the currently plugged plug, but at all `plugOptions`. The 3rd and 4th columns usually only have one perk, but there are weapons with multiple options in these, and weapons with retired perks in the mag slot. I expect this filter to match an item with "Drop Mag" no matter whether it's currently selected or not.

Reply:
Fixed this!